src/passes/SignatureRefining.cpp

```diff
@@ -51,8 +51,8 @@ struct SignatureRefining : public Pass {
     if (!module->features.hasGC()) {
       return;
     }
-    if (getTypeSystem() != TypeSystem::Nominal) {
-      Fatal() << "SignatureRefining requires nominal typing";
+    if (getTypeSystem() == TypeSystem::Equirecursive) {
+      Fatal() << "SignatureRefining requires nominal or isorecursive typing";
```

Comment:
In the `hybrid` branch this and a bunch of other places are updated. The code there does checks for "must be nominal or isorecursive" which I think is more explicit? Sgtm to add one of those checks in this PR but it would be good to have the same code to avoid conflicts later.